packages/mds-compliance-engine/engine/helpers.ts

```diff
@@ -1,38 +1,60 @@
 import cache from '@mds-core/mds-agency-cache'
 import { MatchedVehicleInformation } from '@mds-core/mds-compliance-service'
 import db from '@mds-core/mds-db'
+import { CountPolicy, PolicyDomainModel, Rule, RULE_TYPE, SpeedPolicy, TimePolicy } from '@mds-core/mds-policy-service'
 import { providers } from '@mds-core/mds-providers'
 import {
   DAYS_OF_WEEK,
   DAY_OF_WEEK,
   Device,
   Geography,
-  Policy,
-  Rule,
-  RULE_TYPE,
+  MicroMobilityVehicleEvent,
+  MICRO_MOBILITY_EVENT_STATES_MAP,
+  MICRO_MOBILITY_VEHICLE_EVENTS,
+  MICRO_MOBILITY_VEHICLE_STATE,
+  TaxiVehicleEvent,
+  TAXI_EVENT_STATES_MAP,
+  TAXI_VEHICLE_EVENTS,
+  TAXI_VEHICLE_STATE,
   TIME_FORMAT,
+  TNCVehicleEvent,
+  TNC_EVENT_STATES_MAP,
+  TNC_VEHICLE_EVENT,
+  TNC_VEHICLE_STATE,
   UUID,
   VehicleEvent
 } from '@mds-core/mds-types'
-import { isDefined, now, RuntimeError } from '@mds-core/mds-utils'
+import { areThereCommonElements, isDefined, now, RuntimeError } from '@mds-core/mds-utils'
 import moment from 'moment-timezone'
 import { ProviderInputs, VehicleEventWithTelemetry } from '../@types'
 
 const { env } = process
 
 const TWO_DAYS_IN_MS = 172800000
 
-export function getPolicyType(policy: Policy) {
+export function getPolicyType(policy: PolicyDomainModel) {
   return policy.rules[0].rule_type
 }
 
+export function isCountPolicy(policy: PolicyDomainModel): policy is CountPolicy {
+  return policy.rules[0].rule_type === 'count'
+}
+
+export function isSpeedPolicy(policy: PolicyDomainModel): policy is SpeedPolicy {
+  return policy.rules[0].rule_type === 'speed'
+}
+
+export function isTimePolicy(policy: PolicyDomainModel): policy is TimePolicy {
+  return policy.rules[0].rule_type === 'time'
+}
+
```

Comment:
Maybe out of scope for this PR, but I'd love this to go and run this assertion across all rules. Something along the lines of `export const isCountPolicy = policy.rules.reduce<boolean>((acc, { rule_type }) => acc ? rule_type === 'count' : acc), true)`

Reply:
Or, just `export const isCountPolicy = policy.rules.every(({ rule_type }) => rule_type === ‘count’)` … or even a function that accepts a `rule_type` as a parameter.

Reply:
I forgot that `.every` was a thing! That'd be much more readable than my impl for sure :)